### PR TITLE
[PR-11082] Event webhook + site&type params

### PR DESCRIPTION
**Pull request by human-spider**
## Demo

http://void-docs.talkable.com/

## Related Stories

[](https://talkable.atlassian.net/browse/PR-11082)

### CODE REVIEW

File: source/web_hooks/event.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+       "subscribed_at": "2019-04-11T07:27:31.258-07:00",
+       "opted_in_at": "2019-04-11T07:27:31.258-07:00",
+       "sub_choice": true,
+       "gender": null
+     }
+   }
+
+.. raw:: html
+
+   <h2>cURL example</h2>
```
Comment (zhuravel) [inline source/web_hooks/event.rst]:
```suggestion
   <h2>cURL example | Purchase</h2>
```

File: source/web_hooks/event.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  * **username** — person’s username
+  * **unsubscribed_at** — date person has unsubscribed
+  * **subscribed_at** — date person has subscribed (deprecated; use opted_in_at instead)
+  * **opted_in_at** — date person has subscribed
+  * **sub_choice** — subscription choice
+  * **custom_properties** — hash of person’s custom properties (optional)
+
+.. raw:: html
+
+   <h2>Sample payload | Event</h2>
```
Comment (zhuravel) [inline source/web_hooks/event.rst]: No `cURL example | Event` below.

File: source/web_hooks/event.rst [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    * *"Purchase"* for purchases.
+    * *"Event"* for custom events.
+
+  *For Purchase:*
+
+     * **email** - email address associated with the Purchase
+     * **order_number** - unique identifier of the Purchase
+     * **customer_id** - unique external identifier of a customer who make the purchase
+     * **subtotal** - order subtotal for the purchase
+     * **coupon_code** - unique external identifier of a customer who make the purchase
```
Comment (zhuravel) [inline source/web_hooks/event.rst]: Copy-paste.
```suggestion
     * **coupon_code** - optional coupon code associated with the Purchase
```

File: source/web_hooks/event.rst [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+  *For Event:*
+
+     * **email** - email address associated with the Event
+     * **event_category** - identifier of an action that trigger the Event (e.g. ``app_installed``)
+     * **event_number** - unique identifier of the Event within the associated **event_category**
+     * **customer_id** - unique external identifier of a customer who make the purchase
```
Comment (zhuravel) [inline source/web_hooks/event.rst]:
```suggestion
     * **customer_id** - unique external identifier of a customer who made the purchase
```

File: source/web_hooks/event.rst [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+     * **subtotal** - optional monetary attribute of the Event
+     * **coupon_code** - optional coupon code associated with the Event
+
+* **person** - subhash of data describing the person who triggered the event
+
+  * **first_name** — person’s first name
```
Comment (zhuravel) [inline source/web_hooks/event.rst]: Make sure to correctly merge this with #160.
@mlisivka added a partial for this: https://github.com/talkable/talkable-docs/pull/160/files#diff-685b018f6ef9f5f47c943fb99be74a2f

File: source/web_hooks/event.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+       "unsubscribed_at": null,
+       "subscribed_at": "2019-04-11T07:27:31.258-07:00",
+       "opted_in_at": "2019-04-11T07:27:31.258-07:00",
+       "sub_choice": true
+     }
+   }
+
+.. raw:: html
+
+   <h2>cURL example | Event</h2>
```
Comment (zhuravel) [inline source/web_hooks/event.rst]: This should come right after `Sample payload | Event`.
See how this is done in `web_hooks/reward.rst` for example.

File: source/web_hooks/event.rst [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  * **type**
+
+    * *"Purchase"* for purchases.
+    * *"Event"* for custom events.
+
+  *For Purchase:*
+
+     * **email** - email address associated with the Purchase
+     * **order_number** - unique identifier of the Purchase
+     * **customer_id** - unique external identifier of a customer who make the purchase
```
Comment (zhuravel) [inline source/web_hooks/event.rst]:
```suggestion
     * **customer_id** - unique external identifier of a customer who made the purchase
```

**Comment by human-spider** [comment]
@zhuravel all fixed

**Comment by zhuravel** [comment]
@human-spider Not all, this is not completely fixed:
> Let’s also downcase Purchase in sentences where it does not mean the Ruby class: the Purchase → the purchase.

Change *the Purchase* and *the Event* to *the purchase* and *the event* respectively.
https://github.com/talkable/talkable-docs/pull/161/files#diff-43077d4d5a7964f21fbd0ac6035617e0R38

**Comment by human-spider** [comment]
@zhuravel Purchase is capitalized whenever it's used as Talkable entity as opposed to a purchase made in store. I don't agree we should use lowercase for our entities, even if it doesn't explicitly reference the class.

**Comment by zhuravel** [comment]
@human-spider Fine.

**Pull request merged**